### PR TITLE
OS X support

### PR DESCRIPTION
**Pull request by megawidget**
* sane extensions for Mac
* llvm 8.1 compatibility (older version TensorFlow gets compiled with)

### CODE REVIEW

File: src/ngraph/CMakeLists.txt [hunk 1]
```diff
@@ -345,7 +345,7 @@ endif()
 
 if (APPLE)
     set_property(TARGET ngraph PROPERTY PREFIX "lib")
-    set_property(TARGET ngraph PROPERTY OUTPUT_NAME "ngraph.so")
+    set_property(TARGET ngraph PROPERTY OUTPUT_NAME "ngraph.dylib")
```
Comment (rkimballn1) [inline src/ngraph/CMakeLists.txt]: Not sure what these three lines do, but it looks like they construct the output name. Do we have to use this? What happens if we don't use these three lines?
If you want to use these then let's use them what looks to be the correct way. Set the SUFFIX on line 349 to be ${CMAKE_SHARED_LIBRARY_SUFFIX}. 
I suspect we can remove lines 346 - 350

File: src/ngraph/runtime/backend.cpp [hunk 2]
```diff
@@ -43,7 +43,12 @@ runtime::Backend::~Backend()
 void* runtime::Backend::open_shared_library(const string& type)
 {
     void* handle = nullptr;
-    string name = "lib" + to_lower(type) + "_backend.so";
+    const string name = "lib" + to_lower(type) +
```
Comment (rkimballn1) [inline src/ngraph/runtime/backend.cpp]: Merge in the lastest master to get a better fix for this.

File: test/ngraph.cpp [hunk 3]
```diff
@@ -29,7 +29,8 @@ using namespace std;
 TEST(NGraph, loadTest)
 {
     // load the triangle library
-    void* ngraphImplLib = dlopen("../src/libngraph.so", RTLD_LAZY);
+    void* ngraphImplLib = dlopen("../src/libngraph.so", RTLD_LAZY) ||
```
Comment (rkimballn1) [inline test/ngraph.cpp]: We could do this the same way it is done in backend.cpp, that way it supports all targets.

**Comment by megawidget** [comment]
needs backmerge from master, blocked by #1112

File: src/ngraph/runtime/backend.cpp (hunk 2 above)
Comment (rkimballn1) [reply src/ngraph/runtime/backend.cpp]: Can you just pull this file from master. The code on master supports linux, macos and windows.

**Pull request merged**